zeroex/orderwatch/order_watcher.go

```diff
@@ -1011,6 +1008,62 @@ func (w *Watcher) trimOrdersAndGenerateEvents() ([]*zeroex.OrderEvent, error) {
 	return orderEvents, nil
 }
 
+// updateBlockHeadersStoredInDB updates the block headers stored in the DB. Since our DB txns don't support
+// multiple operations involving the same entry, we make sure we only perform either an insertion or a deletion
+// for each block in this method.
+func (w *Watcher) updateBlockHeadersStoredInDB(miniHeadersColTxn *db.Transaction, events []*blockwatch.Event) error {
```

Comment:
It looks like we don't actually use `w` inside this method. If we change this from a method to a function, it will be a lot easier to test. We wouldn't have to set up any blockchain state and it could be run in parallel (as long as we use a randomly generated name for the db directory).